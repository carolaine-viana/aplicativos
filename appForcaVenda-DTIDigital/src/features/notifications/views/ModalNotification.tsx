import React from 'react';

import { ModalPopUp } from '@shared/components';
import { Container, ModalTitle, ModalDescription } from './ModalNotificationSC';

interface Props {
  modalVisible: boolean;
  closeModal: () => void;
}

function ModalNotification({ modalVisible, closeModal }: Props) {
  return (
    <ModalPopUp visible={modalVisible} onCloseModal={closeModal} isNotification>
      <Container>
        <ModalTitle color="primary" variant="titleLarge">
          Title
        </ModalTitle>
        <ModalDescription color="default" variant="overline" size={14} numberOfLines={2}>
          Title Title Title Title Title Title Title Title Title Title Title Title Title Title Title
          Title Title Title
        </ModalDescription>
      </Container>
    </ModalPopUp>
  );
}
export default ModalNotification;
