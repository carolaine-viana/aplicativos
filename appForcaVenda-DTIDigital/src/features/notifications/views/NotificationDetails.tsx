import React from 'react';

import { useNavigation, useRoute } from '@react-navigation/native';
import { CustomStatusBar, SearchHeader, Typography, DefaultIcon } from '@shared/components';
import { useTheme } from 'styled-components';
import {
  Container,
  Title,
  Description,
  DownloadButton,
  ContainerButton,
} from './NotificationDetailsSC';

const NotificationDetails: React.FC = () => {
  const navigation = useNavigation();
  const theme = useTheme();
  const route = useRoute();

  return (
    <>
      <CustomStatusBar backgroundColor={theme.palette.blue50} barStyle="dark-content" />
      <SearchHeader
        headerProps={{ title: 'Notificações' }}
        onClickArrow={() => navigation.goBack()}
      />
      <Container>
        <Title size={18} color="gray600" variant="bodyMedium">
          {route?.params?.title}
        </Title>
        <Description size={11} color="gray700" variant="overline">
          {route?.params?.description}
        </Description>

        {route?.params?.file && (
          <ContainerButton>
            <DownloadButton onPress={() => {}}>
              <DefaultIcon name="paperclip" size={24} color="light" />
              <Typography variant="bodyMedium" color="white">
                Baixar anexo
              </Typography>
            </DownloadButton>
          </ContainerButton>
        )}
      </Container>
    </>
  );
};

export default NotificationDetails;
