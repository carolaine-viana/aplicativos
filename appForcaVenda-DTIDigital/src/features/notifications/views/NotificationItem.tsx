import React from 'react';
import { Divider, DefaultIcon } from '@shared/components';

import {
  Container,
  Title,
  Description,
  ContainerDetails,
  Date,
  Hour,
  ContainerIcon,
} from './NotificationItemSC';

interface ItemProps {
  title: string;
  description: string;
  date: string;
  hour: string;
  file?: boolean;
}
interface NotificationItemProps {
  item: ItemProps;
  onPress: () => void;
}
const NotificationItem: React.FC<NotificationItemProps> = ({ item, onPress }) => (
  <Container onPress={onPress}>
    <Title size={14} variant="bodyMedium" color="gray700">
      {item.title}
    </Title>
    <Description size={11} variant="overline" color="gray700" numberOfLines={2}>
      {item.description}
    </Description>
    <ContainerDetails>
      <Date size={9} color="gray300">
        {item.date}
      </Date>
      <Hour size={9} color="gray300">
        {item.hour}
      </Hour>
      {item.file && (
        <ContainerIcon>
          <DefaultIcon name="paperclip" size={12} color="link" />
        </ContainerIcon>
      )}
    </ContainerDetails>
    <Divider color="blue100" />
  </Container>
);

export default NotificationItem;
