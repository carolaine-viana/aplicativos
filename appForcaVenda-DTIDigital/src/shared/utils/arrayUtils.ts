/**
 * Use this to deal/manipulate arrays
 */
export function isArrayEmpty<T = null>(arr: Array<T> | null | undefined) {
  return arr && arr?.length === 0;
}

export function doesArrayHaveContent<T = null>(arr: Array<T> | null | undefined) {
  const isEmpty = isArrayEmpty(arr);
  if (isEmpty) {
    return false;
  }
  return arr?.some((i) => !!i);
}

export function filterArraObjectAnyValue<T>(
  arr: Array<T>,
  valueFilter: string,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  keysFilter?: Array<keyof T>
) {
  return arr.filter((item: T) => {
    const values = Object.values(item).map((value: string) => {
      if (typeof value !== 'object') {
        return String(value).toUpperCase();
      }
      return null;
    });
    return values.join(' ').includes(valueFilter.toUpperCase());
  });
}

export const addNoRepeatedString = (list: string[], newValue: string) => {
  const newList = list.filter((str) => str !== newValue || str.length === 0);
  newList.push(newValue);
  return newList;
};

export const mergArrayIntoStrings = (list: string[], separator = ';') => {
  if (!list || isArrayEmpty(list)) {
    return '';
  }

  const noEmptyStrings = list.filter((str) => str.length > 0);
  const mergedString =
    noEmptyStrings.length === 1 ? noEmptyStrings[0] : noEmptyStrings.join(separator);

  return mergedString;
};
