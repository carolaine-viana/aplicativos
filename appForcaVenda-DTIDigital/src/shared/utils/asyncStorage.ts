import AsyncStorage from '@react-native-community/async-storage';
import { Profiles, TRole } from '@shared/types';
import { addNoRepeatedString, getISOStringDate, mergArrayIntoStrings } from '.';

export const FV_TOKEN = '@fv/token';
export const FV_TOKEN_EXPIRATION = '@fv/tokenExpiration';
export const FV_USER_LOGIN = '@fv/rcaLogin';
export const FV_USER_ROLE = '@fv/userRole';
export const FV_FIRST_TUTO_MAP = '@fv/firstTutoMap';
export const FV_FEATURE_FLAGS = '@fv/featureFlags';
export const FV_COMPLETED_TUTORIALS = '@fv/completedTutorials';
export const FV_LAST_TIME_USER_SAW_TUTORIAL = '@fv/lastTimeUserSawTutorial';
export const FV_DATE_ACCEPTED_TERM_USE = '@fv/dateAcceptedTermUse';
export const FV_DATE_UPDATED_TERM_USE = '@fv/dateUpdatedTermUse';

export const clear = () => AsyncStorage.clear();

// TOKEN INFO
export const setToken = async (token: string) => {
  await AsyncStorage.setItem(FV_TOKEN, token);
};

export const getToken = async () => {
  const token = await AsyncStorage.getItem(FV_TOKEN);
  return token;
};

export const setTokenExpiration = async (expirationDate: Date) => {
  await AsyncStorage.setItem(FV_TOKEN_EXPIRATION, expirationDate.toString());
};

export const getTokenExpiration = async (): Promise<Date | number> => {
  const expirationTime = await AsyncStorage.getItem(FV_TOKEN_EXPIRATION);
  return expirationTime ? new Date(expirationTime) : 0;
};

// USER INFO
export const getUserLogin = async () => {
  const userLogin = await AsyncStorage.getItem(FV_USER_LOGIN);
  return userLogin;
};

export const setUserLogin = async (login: string) => {
  await AsyncStorage.setItem(FV_USER_LOGIN, login);
};

export const getUserRole = async (): Promise<TRole | null> => {
  const userRole = await AsyncStorage.getItem(FV_USER_ROLE);
  return userRole as TRole;
};

export const setUserRole = async (role: TRole) => {
  await AsyncStorage.setItem(FV_USER_ROLE, role);
};

export const isUserGV = async (): Promise<boolean> => {
  const userRole = await getUserRole();
  return userRole === Profiles.GV;
};

// FEATURE FLAG
export const setFeatureFlags = async (featureFlags: string) => {
  await AsyncStorage.setItem(FV_FEATURE_FLAGS, featureFlags);
};

export const getFeatureFlags = async () => {
  const flags = await AsyncStorage.getItem(FV_FEATURE_FLAGS);
  return flags || '';
};

// TUTORIALS
export const getCompletedTutorials = async () => {
  const completedTutorials = await AsyncStorage.getItem(FV_COMPLETED_TUTORIALS);
  return completedTutorials ?? '';
};

export const setCompletedTutorials = async (tutorial: string) => {
  const completedTutorials = await getCompletedTutorials();
  if (completedTutorials !== ';') {
    const newListCompletedTutorials = completedTutorials.split(';');
    const uniqueStrings = addNoRepeatedString(newListCompletedTutorials, tutorial);
    await AsyncStorage.setItem(FV_COMPLETED_TUTORIALS, mergArrayIntoStrings(uniqueStrings));
  }
};

export const getLastTimeUserSawTutorial = () =>
  AsyncStorage.getItem(FV_LAST_TIME_USER_SAW_TUTORIAL);

export const setLastTimeUserSawTutorial = () => {
  const isoStringDate = getISOStringDate(new Date());
  AsyncStorage.setItem(FV_LAST_TIME_USER_SAW_TUTORIAL, isoStringDate);
};

export const getFirstTutoMap = async (visualized?: string) => {
  if (visualized === 'visualized') {
    await AsyncStorage.setItem(FV_FIRST_TUTO_MAP, 'visualized');
  }
  const firstMap = await AsyncStorage.getItem(FV_FIRST_TUTO_MAP);
  return firstMap;
};

// PRIVACY POLICY
export const getAcceptedTermUseDate = () => AsyncStorage.getItem(FV_DATE_ACCEPTED_TERM_USE);

export const setDateAcceptedTermUse = (lastDate: string | undefined) => {
  if (lastDate) {
    return AsyncStorage.setItem(FV_DATE_ACCEPTED_TERM_USE, lastDate);
  }
};

export const getUpdatedTermUseDate = () => AsyncStorage.getItem(FV_DATE_UPDATED_TERM_USE);
export const setDateUpdatedTermUse = (updatedDate: string | undefined) => {
  if (updatedDate) {
    return AsyncStorage.setItem(FV_DATE_UPDATED_TERM_USE, updatedDate);
  }
};
