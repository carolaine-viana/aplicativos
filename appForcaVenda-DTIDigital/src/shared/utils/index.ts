export * from './generalUtils';
export * from './dateUtils';
export * from './arrayUtils';
export * from './mappers/mapperCustomerVisit';
