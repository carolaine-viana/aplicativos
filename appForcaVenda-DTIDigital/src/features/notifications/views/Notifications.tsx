import React, { useState } from 'react';

import { useNavigation } from '@react-navigation/native';
import { useTheme } from 'styled-components';
import { CustomStatusBar, SearchHeader } from '@shared/components';
import { RoutesNames } from 'shared/types';
import NotificationItem from './NotificationItem';
import { Container } from './NotificationsSC';

const item = {
  title: 'AQUI é o TITLE',
  description:
    'Descrição Descrição Descrição Descrição Descrição Descrição Descrição Descrição DescriçãoDescrição Descrição Descrição Descrição Descrição Descrição Descrição',
  date: '01/09/2022',
  hour: '15:00h',
  file: true,
};
const Notifications: React.FC = () => {
  const theme = useTheme();
  const navigation = useNavigation();
  const [search, setSearch] = useState('');

  const handleCloseSearch = () => {
    setTimeout(() => {
      setSearch('');
    }, 200);
  };

  return (
    <>
      <CustomStatusBar backgroundColor={theme.palette.blue50} barStyle="dark-content" />
      <SearchHeader
        headerProps={{ title: 'Notificações' }}
        onClickArrow={() => navigation.goBack()}
        searchProps={{
          onChangeText: (text: string) => {
            setSearch(text);
          },
          onClose: handleCloseSearch,
          value: search,
          searchPlaceholder: 'Buscar Notificação',
        }}
      />
      <Container>
        <NotificationItem
          item={item}
          onPress={() =>
            navigation.navigate(RoutesNames.NOTIFICATION_DETAILS, {
              title: item.title,
              file: item.file,
              description: item.description,
            })
          }
        />
      </Container>
    </>
  );
};

export default Notifications;
