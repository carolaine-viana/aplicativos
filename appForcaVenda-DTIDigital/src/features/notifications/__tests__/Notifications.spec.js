import React from 'react';
import expect from 'expect';
import { fireEvent, render, waitFor, act } from '@shared/utils/test-utils';
import NotificationDetails from '../views/NotificationDetails'
import NotificationItem from '../views/NotificationItem'
import Notifications from '../views/Notifications'
import ModalNotification from '../views/ModalNotification'
import jestExpect from 'expect';

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
  useRoute: jest.fn(),
}));

jest.mock('styled-components', () => {
  return {
    useTheme: jest.fn().mockReturnValue({
      palette: {
        white: '#ffffff',
      },
    }),
  };
});
it('should be render Notifications correctly', () => {
  const screen = render(<Notifications />).toJSON();
  jestExpect(screen).toMatchSnapshot();
})

it('should be render Notifications NotificationItem', () => {
  const screen = render(<NotificationItem item={{title: 'Titulo', description: 'DescriÃ§ao', date: '01/09/2022', hour: '15h', file: true}} />).toJSON();
  jestExpect(screen).toMatchSnapshot();
})

it('should be render Notifications NotificationDetails', () => {
  const screen = render(<NotificationDetails />).toJSON();
  jestExpect(screen).toMatchSnapshot();
})

it('should be render Notifications ModalNotification', () => {
  const screen = render(<ModalNotification />).toJSON();
  jestExpect(screen).toMatchSnapshot();
})


