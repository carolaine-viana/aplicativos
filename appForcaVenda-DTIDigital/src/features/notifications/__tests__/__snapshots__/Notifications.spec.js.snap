// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be render Notifications ModalNotification 1`] = `
<View
  customStyles={
    Object {
      "backdropColor": "rgba(0,0,0,0.7)",
      "modalBorderRadius": 4,
      "positionTop": 0,
    }
  }
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "height": 1334,
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 0,
        "position": "relative",
        "top": 0,
        "width": "100%",
        "zIndex": 3,
      },
    ]
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "transform": Array [
          Object {
            "scale": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginBottom": 10,
            "marginLeft": -10,
            "marginRight": -10,
            "marginTop": 0,
            "zIndex": 5,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#3289FF",
              "borderRadius": 23,
              "height": 46,
              "justifyContent": "center",
              "position": "absolute",
              "width": 46,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 26,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      customStyles={
        Object {
          "backdropColor": "rgba(0,0,0,0.7)",
          "modalBorderRadius": 4,
          "positionTop": 0,
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 30,
            "elevation": 20,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "zIndex": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "width": "100%",
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "minHeight": 200,
                "paddingBottom": 117,
                "paddingLeft": 57,
                "paddingRight": 57,
                "paddingTop": 117,
              },
            ]
          }
        >
          <Text
            color="primary"
            fallbackText=""
            style={
              Array [
                Object {
                  "color": "#072AD8",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 24,
                  "lineHeight": 28,
                  "textTransform": "none",
                },
                Object {
                  "marginBottom": 14,
                },
              ]
            }
            variant="titleLarge"
          >
            Title
          </Text>
          <Text
            color="default"
            fallbackText=""
            numberOfLines={2}
            size={14}
            style={
              Array [
                Object {
                  "color": "#10151D",
                  "fontFamily": "Rubik-Regular",
                  "fontSize": 14,
                  "lineHeight": 13,
                  "textTransform": "none",
                },
                Object {},
              ]
            }
            variant="overline"
          >
            Title Title Title Title Title Title Title Title Title Title Title Title Title Title Title Title Title Title
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`should be render Notifications NotificationDetails 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "",
          "height": "px",
          "zIndex": 3,
        },
      ]
    }
    translucent={true}
  />,
  <View
    backgroundColor="default"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "height": 49,
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 1.5,
            "width": 1.5,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1.5,
          "width": "100%",
          "zIndex": 2,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 0,
            "zIndex": 2,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 36,
            "justifyContent": "center",
            "opacity": 1,
            "width": 36,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#072AD8",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <Text
      bold={true}
      color="gray400"
      fallbackText=""
      size={18}
      style={
        Array [
          Object {
            "color": "#838896",
            "fontFamily": "Rubik-Bold",
            "fontSize": 18,
            "lineHeight": 28,
            "textTransform": "none",
          },
          Object {
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "textAlign": "center",
          },
        ]
      }
      variant="header"
    >
      Notificações
    </Text>
    <View
      style={
        Array [
          Object {
            "right": 0,
          },
        ]
      }
    />
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 30,
        },
      ]
    }
  >
    <Text
      color="gray600"
      fallbackText=""
      size={18}
      style={
        Array [
          Object {
            "color": "#595F70",
            "fontFamily": "Rubik-Medium",
            "fontSize": 18,
            "lineHeight": 17,
            "textTransform": "none",
          },
          Object {
            "lineHeight": 22,
            "marginBottom": 12,
          },
        ]
      }
      variant="bodyMedium"
    >
      
    </Text>
    <Text
      color="gray700"
      fallbackText=""
      size={11}
      style={
        Array [
          Object {
            "color": "#3A4050",
            "fontFamily": "Rubik-Regular",
            "fontSize": 11,
            "lineHeight": 13,
            "textTransform": "none",
          },
          Object {
            "lineHeight": 13,
            "marginBottom": 38,
          },
        ]
      }
      variant="overline"
    >
      
    </Text>
  </View>,
]
`;

exports[`should be render Notifications NotificationItem 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "paddingBottom": 24,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 24,
      },
    ]
  }
>
  <Text
    color="gray700"
    fallbackText=""
    size={14}
    style={
      Array [
        Object {
          "color": "#3A4050",
          "fontFamily": "Rubik-Medium",
          "fontSize": 14,
          "lineHeight": 17,
          "textTransform": "none",
        },
        Object {
          "lineHeight": 17,
          "marginBottom": 8,
        },
      ]
    }
    variant="bodyMedium"
  >
    Titulo
  </Text>
  <Text
    color="gray700"
    fallbackText=""
    numberOfLines={2}
    size={11}
    style={
      Array [
        Object {
          "color": "#3A4050",
          "fontFamily": "Rubik-Regular",
          "fontSize": 11,
          "lineHeight": 13,
          "textTransform": "none",
        },
        Object {
          "lineHeight": 13,
          "marginBottom": 8,
        },
      ]
    }
    variant="overline"
  >
    Descriçao
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": -8,
        },
      ]
    }
  >
    <Text
      color="gray300"
      fallbackText=""
      size={9}
      style={
        Array [
          Object {
            "color": "#999DA9",
            "fontFamily": "Rubik-Regular",
            "fontSize": 9,
            "lineHeight": 28,
            "textTransform": "none",
          },
          Object {
            "marginRight": 8,
          },
        ]
      }
      variant="header"
    >
      01/09/2022
    </Text>
    <Text
      color="gray300"
      fallbackText=""
      size={9}
      style={
        Array [
          Object {
            "color": "#999DA9",
            "fontFamily": "Rubik-Regular",
            "fontSize": 9,
            "lineHeight": 28,
            "textTransform": "none",
          },
          Object {},
        ]
      }
      variant="header"
    >
      15h
    </Text>
    <View
      style={
        Array [
          Object {
            "marginLeft": 6,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#3289FF",
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    noMarge={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "marginTop": 16,
          "width": "100%",
        },
      ]
    }
  >
    <View
      color="blue100"
      style={
        Array [
          Object {
            "borderBottomColor": "#B3D3FF",
            "borderBottomWidth": 1,
            "width": "100%",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`should be render Notifications correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "backgroundColor": "",
          "height": "px",
          "zIndex": 3,
        },
      ]
    }
    translucent={true}
  />,
  <View
    backgroundColor="default"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flexDirection": "row",
          "height": 49,
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 0,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 1.5,
            "width": 1.5,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1.5,
          "width": "100%",
          "zIndex": 2,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 0,
            "zIndex": 2,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 36,
            "justifyContent": "center",
            "opacity": 1,
            "width": 36,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#072AD8",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <Text
      bold={true}
      color="gray400"
      fallbackText=""
      size={18}
      style={
        Array [
          Object {
            "color": "#838896",
            "fontFamily": "Rubik-Bold",
            "fontSize": 18,
            "lineHeight": 28,
            "textTransform": "none",
          },
          Object {
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "textAlign": "center",
          },
        ]
      }
      variant="header"
    >
      Notificações
    </Text>
    <View
      style={
        Array [
          Object {
            "right": 0,
          },
        ]
      }
    >
      <View
        collapsable={false}
        color="gray200"
        isSearchVisible={false}
        nativeID="animatedComponent"
        style={
          Object {
            "backgroundColor": "#D3D5DA",
            "borderRadius": 30,
            "height": 36,
            "width": 40,
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Buscar Notificação"
          placeholderTextColor="#595F70"
          style={
            Array [
              Object {
                "color": "#000",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginLeft": 20,
                "marginRight": 60,
              },
            ]
          }
          testID="searchTextIpt"
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "height": 36,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "right": 6,
            "width": 36,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#072AD8",
                  "fontSize": 24,
                },
              ],
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "paddingBottom": 24,
            "paddingLeft": 24,
            "paddingRight": 24,
            "paddingTop": 24,
          },
        ]
      }
    >
      <Text
        color="gray700"
        fallbackText=""
        size={14}
        style={
          Array [
            Object {
              "color": "#3A4050",
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
              "lineHeight": 17,
              "textTransform": "none",
            },
            Object {
              "lineHeight": 17,
              "marginBottom": 8,
            },
          ]
        }
        variant="bodyMedium"
      >
        AQUI é o TITLE
      </Text>
      <Text
        color="gray700"
        fallbackText=""
        numberOfLines={2}
        size={11}
        style={
          Array [
            Object {
              "color": "#3A4050",
              "fontFamily": "Rubik-Regular",
              "fontSize": 11,
              "lineHeight": 13,
              "textTransform": "none",
            },
            Object {
              "lineHeight": 13,
              "marginBottom": 8,
            },
          ]
        }
        variant="overline"
      >
        Descrição Descrição Descrição Descrição Descrição Descrição Descrição Descrição DescriçãoDescrição Descrição Descrição Descrição Descrição Descrição Descrição
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": -8,
            },
          ]
        }
      >
        <Text
          color="gray300"
          fallbackText=""
          size={9}
          style={
            Array [
              Object {
                "color": "#999DA9",
                "fontFamily": "Rubik-Regular",
                "fontSize": 9,
                "lineHeight": 28,
                "textTransform": "none",
              },
              Object {
                "marginRight": 8,
              },
            ]
          }
          variant="header"
        >
          01/09/2022
        </Text>
        <Text
          color="gray300"
          fallbackText=""
          size={9}
          style={
            Array [
              Object {
                "color": "#999DA9",
                "fontFamily": "Rubik-Regular",
                "fontSize": 9,
                "lineHeight": 28,
                "textTransform": "none",
              },
              Object {},
            ]
          }
          variant="header"
        >
          15:00h
        </Text>
        <View
          style={
            Array [
              Object {
                "marginLeft": 6,
              },
            ]
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#3289FF",
                  "fontSize": 12,
                },
                undefined,
                Object {
                  "fontFamily": "Feather",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        noMarge={false}
        style={
          Array [
            Object {
              "alignItems": "center",
              "marginTop": 16,
              "width": "100%",
            },
          ]
        }
      >
        <View
          color="blue100"
          style={
            Array [
              Object {
                "borderBottomColor": "#B3D3FF",
                "borderBottomWidth": 1,
                "width": "100%",
              },
            ]
          }
        />
      </View>
    </View>
  </View>,
]
`;
